network/src/p2p.rs

```diff
@@ -296,3 +317,43 @@ impl ReliableNetwork<WorkerMessage> for P2pNetwork {
         self.send(peer, f).await
     }
 }
+
+#[async_trait]
+impl UnreliableNetwork<WorkerBatchRequest> for P2pNetwork {
+    type Response = WorkerBatchResponse;
+    async fn unreliable_send(
+        &mut self,
+        peer: NetworkPublicKey,
+        message: &WorkerBatchRequest,
+    ) -> JoinHandle<anyhow::Result<anemo::Response<WorkerBatchResponse>>> {
+        let message = message.to_owned();
+        let f = move |peer| async move {
+            WorkerToWorkerClient::new(peer)
+                .request_batches(message)
+                .await
+        };
+        self.unreliable_send(peer, f).await
+    }
+}
+
+#[async_trait]
+impl ReliableNetwork<WorkerBatchRequest> for P2pNetwork {
+    type Response = WorkerBatchResponse;
+    async fn send(
+        &mut self,
+        peer: NetworkPublicKey,
+        message: &WorkerBatchRequest,
+    ) -> CancelOnDropHandler<anyhow::Result<anemo::Response<WorkerBatchResponse>>> {
+        let message = message.to_owned();
+        let f = move |peer| {
+            let message = message.clone();
```

Comment:
Why another clone() is needed?

Reply:
It's cloned so it can be moved into the async block. `send` needs to be able to call this closure multiple times in case of retry so it has to be `Fn` and not just `FnOnce`